File: solutions/haskell/series/src/Series.hs
module Series (slices) where

import Data.Char as Char

charToInt :: Char -> Int
charToInt c = Char.ord c - Char.ord '0'

substring :: Int -> Int -> String -> String
substring start end = take (end - start + 1) . drop start

slices :: Int -> String -> [[Int]]
slices n xs = [map charToInt (substring i (i + n - 1) xs) | i <- [0..(length xs) - n]]

File: solutions/csharp/series/Series.cs
public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength <= 0 || sliceLength > numbers.Length)
            throw new ArgumentException("");

        var slices = new string[numbers.Length - sliceLength + 1];

        for (int i = 0; i < slices.Length; i++)
            slices[i] = numbers.Substring(i, sliceLength);

        return slices;
    }
}